invoice-service/src/main/java/com/registry/invoiceservice/service/InvoiceService.java

```diff
@@ -15,21 +20,61 @@
     @Autowired
     private InvoiceRepository invoiceRepository;
 
-    public Invoice createInvoice(Invoice invoice){
+    public InvoiceDTO createInvoice(Invoice invoice) {
         List<Item> itemList = invoice.getItems();
-        for(Item item: itemList){
-            item.setTotalPrice(item.getSinglePrice() + item.getQuantity()*item.getVat());
+        for (Item item : itemList) {
+            item.setTotalPrice(item.getSinglePrice().add(item.getQuantity().multiply(item.getVat())));
         }
         invoice.setItems(itemList);
-        return invoiceRepository.save(invoice);
+        return mapToInvoiceDTO(invoiceRepository.save(invoice));
     }
 
-    public List<Invoice> getAllInvoices(){
-        return invoiceRepository.findAll();
+    public Page<Invoice> getAllInvoices(Integer page, Integer size) {
+        if (page == null || size == null) {
+            return invoiceRepository.findAll(PageRequest.of(0, (int) invoiceRepository.count()));
+        }
+        return invoiceRepository.findAll(PageRequest.of(page, size));
+    }
+
+    public InvoiceDTO getInvoiceById(Long id) {
+        return mapToInvoiceDTO(invoiceRepository.findById(id).get());
     }
 
-    public Invoice getInvoiceById(Integer id){
-        return invoiceRepository.findById(id).orElseThrow(NullPointerException::new);
+    public List<InvoiceDTO> getInvoicesByCriteria(CriteriaDto criteriaDto) {
+        List<InvoiceDTO> invoiceDTOS = new ArrayList<>();
+        List<Invoice> invoices = invoiceRepository.searchInvoiceByCriteria(criteriaDto.getQuantityGreaterThan(), criteriaDto.getTotalPriceGreaterThan());
+        for(Invoice invoice : invoices){
+            invoiceDTOS.add(mapToInvoiceDTO(invoice));
+        }
+        return  invoiceDTOS;
+    }
+
+    private ItemDTO mapToItemDTO(Item item){
+        ItemDTO itemDTO = new ItemDTO();
+        itemDTO.setName(item.getName());
+        itemDTO.setQuantity(item.getQuantity());
+        itemDTO.setCatalogueNumber(item.getCatalogueNumber());
+        itemDTO.setTotalPrice(item.getTotalPrice());
+        itemDTO.setSerialNumber(item.getSerialNumber());
+        itemDTO.setSinglePrice(item.getSinglePrice());
+        itemDTO.setVat(item.getVat());
+        return itemDTO;
+    }
+
+    private InvoiceDTO mapToInvoiceDTO(Invoice invoice){
```

Comment:
If we have time (in terms of training plan) we should look into mapping libraries like MapStruct.